packages/edit-site/src/components/page-patterns/index.js

```diff
@@ -337,27 +333,12 @@ export default function DataviewsPatterns() {
 		}
 	}, [ categoryId, previousCategoryId ] );
 	const { data, paginationInfo } = useMemo( () => {
-		if ( ! patterns ) {
-			return {
-				data: EMPTY_ARRAY,
-				paginationInfo: { totalItems: 0, totalPages: 0 },
-			};
-		}
-		let filteredData = [ ...patterns ];
-		// Handle sorting.
-		if ( view.sort ) {
-			filteredData = sortByTextFields( {
-				data: filteredData,
-				view,
-				fields,
-				textFields: [ 'title', 'author' ],
-			} );
-		}
-		// Handle pagination.
-		return getPaginationResults( {
-			data: filteredData,
-			view,
-		} );
+		// Since filters are applied server-side,
```

Comment:
This is an optimization, as it should work with the filters & search. Not super fan of asking consumers to do this, I'd like it more if:

- offer a config for this all-in-one utility, as in `filterAndSortDataView( data, view, fields, options );`
- create micro-utilities (`sort`, `filter`, `paginate`)

Reply:
I don't like micro-utilities personally for this but I'd be open for options. Maybe we can start like today and iterate.

Reply:
What would you think of adding an `options` parameter?

Reply:
To be honest, that was the first thing I did when I was working on the PR but then I though, why adding an "filter" flag and at the same time having "filters" in view. "view" in a sense is already an options param, so I just deleted the filters. 

I guess this is me saying that I'm not sure what's the best. I agree that for us as consumers it's simpler to have an option but if you think about the API itself in isolation, it feels weird to pass "filters" but to tell it not to filter.

Reply:
Yeah, I see.

Other libraries absorb this complexity (IIRC, tanstack table had configuration for that) but I actually like the simpler data flow we have: `data` is received by `DataViews` already prepared. If a consumer forgets to remove the operations the worst that will happen is that it becomes a little more slow, probably isn't going to be a big deal. We could add some comments/docs about this use case.